sklearn/metrics/pairwise.py

```diff
@@ -155,6 +178,15 @@ def check_pairwise_arrays(
         An array equal to Y if Y was not None, guaranteed to be a numpy array.
         If Y was None, safe_Y will be a pointer to X.
     """
+    if force_all_finite != "deprecated":
+        warnings.warn(
+            "'force_all_finite' was renamed to 'ensure_all_finite' in 1.6 and will be "
+            "removed in 1.8. Until then, ensure_all_finite is ignored when "
+            "force_all_finite is set.",
+            FutureWarning,
+        )
```

Comment:
usually in these cases we raise if both are set. Only the new one should be set. More like

```py
if force_all_finite != "deprecated" and ensure_all_finate != "default":
  raise ...
elif force_all_finite != "deprecated":
  warn("use ensure...")
  ensure_... = force_...
elif ensure_... == "default":
  ensure_... = True
```

Reply:
See the discussion here https://github.com/scikit-learn/scikit-learn/pull/29404#discussion_r1669502128, and especially my last comment https://github.com/scikit-learn/scikit-learn/pull/29404#discussion_r1680700979. I agree with you, but I just want to make sure that we're on the same page regarding the removal of "default" or None (whichever we chose) in 1.8 without deprecation.